op-challenger/game/fault/solver/game_solver_test.go

```diff
@@ -84,6 +85,32 @@ func TestCalculateNextActions(t *testing.T) {
 				lastHonestClaim.Attack(common.Hash{0xdd}).ExpectStepAttack()
 			},
 		},
+
+		{
+			name:                "PoisonedPreState",
+			agreeWithOutputRoot: true,
+			setupGame: func(builder *faulttest.GameBuilder) {
+				// Create a cannon state that is exited and resolved that the output root was invalid
+				maliciousState := mipsevm.State{
+					Memory:   mipsevm.NewMemory(),
+					ExitCode: 1,
+					Exited:   true,
+				}
+				maliciousStateHash, _ := maliciousState.EncodeWitness().StateHash()
+
+				// Dishonest actor counters their own claims to set up a situation with an invalid prestate
+				// The honest actor should attack all claims that support the root claim (disagree with the output root)
+				builder.Seq().ExpectAttack(). // This expected action is the winning move.
+								Attack(maliciousStateHash).
+								Defend(maliciousStateHash).ExpectAttack().
+								Attack(maliciousStateHash).
+								Attack(maliciousStateHash).ExpectStepAttack()
+
+				// The attempt to step against our malicious leaf node will fail because the pre-state won't match our
+				// malicious state hash. However, it is the very first expected action, attacking the root claim with
+				// the correct hash that wins the game since it will be the left-most uncountered claim.
```

Comment:
Isn't the attack against root also malicious, with an incorrect hash?
Seems to me that this game, the CHALLENGER wins (by default) not because the first action is correct, but because the DEFENDER failed provide an uncontested claim. Which is the correct outcome.

Reply:
There will be two attacks against the root - the malicious one from the first `Attack(maliciousStateHash)` call which is already countered by the dishonest actor and we expect the honest actor to also attack the root but with the correct claim.  The honest actor's attack on the root will be what goes on to win the game - the dishonest actor could respond to it but it shows the honest actor can "escape" the poisoning by attacking left.

Expected moves are:
```
    game_solver_test.go:125: Move 0: Type: move, ParentIdx: 0, Attack: true, Value: 0x0157c6880317692d1e241d4e588624849a414af4c7f958811b533447ec42aec8, PreState: , ProofData: 
    game_solver_test.go:125: Move 1: Type: move, ParentIdx: 2, Attack: true, Value: 0x0110caf4eb645e6e135f3034a51a2295a2e216f2fb35fdb47d2c2a2d591db7ff, PreState: , ProofData: 
    game_solver_test.go:125: Move 2: Type: step, ParentIdx: 4, Attack: true, Value: 0x0000000000000000000000000000000000000000000000000000000000000000, PreState: 00000000000000000000000000000000000000000000000000000000000000070000000000000000000000000000000000000000000000000000000000000068, ProofData: 07
```

Reply:
gotcha. I misread the setup code as the only moves in the game.